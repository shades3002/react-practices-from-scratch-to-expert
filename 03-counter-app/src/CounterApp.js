import React, { Fragment, useState } from 'react';
import PropTypes from 'prop-types';

// use snippet rafcp 

const CounterApp = ({ value = 0 }) => {

    const [count, setCount] = useState(value);

    const handAdd = (e) => {
        setCount(count + 1);
        //setCount((c) => c + 1);
        console.log('event click', count);
    }

    const handReset = (e) => {
        setCount(value);
        console.log('event click', count);
    }

    const handSubtract = (e) => {
        setCount(count - 1);
        console.log('event click', count);
    }

    return (
        <Fragment>
            <h1>CounterApp</h1>
            <h2>{ count }</h2>

            {/* <button onClick={(e) => {
                console.log('event click');
                value++;
                console.log(value);
            }}>+1</button>   */}

            <button onClick={ handAdd }>+1</button>
            <button onClick={ handReset }>Reset</button>
            <button onClick={ handSubtract }>-1</button>
        </Fragment>
    );
}
 
CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

export default CounterApp;