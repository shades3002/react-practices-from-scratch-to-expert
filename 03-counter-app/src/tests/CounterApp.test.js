import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import CounterApp from '../CounterApp';



describe('Test component <CounterApp />', ()=> {

    let wrapper = shallow( <CounterApp /> );
    
    beforeEach( () => {
        wrapper = shallow( <CounterApp /> );
    });

    //const wrapper = shallow( <CounterApp /> );

    test('should show <CounterApp /> correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });


    test('should counter correctly', () => {
        const wrapper = shallow( 
            <CounterApp 
                value={ 100 } 
            /> 
        );

        const counter = wrapper.find('h2').text();

        expect( Number(counter) ).toBe( 100 );

        expect( counter.trim() ).toBe( '100' );
    });


        
    test('should action click Btn +1', () => {
        wrapper.find('button').at(0).simulate('click');

        const counter = wrapper.find('h2').text().trim();

        expect( counter ).toBe( '1' );

    });

    test('should action click Btn -1', () => {
        wrapper.find('button').at(2).simulate('click');

        const counter = wrapper.find('h2').text().trim();

        expect( counter ).toBe( '-1' );

    });

    test('should action click Btn Reset', () => {
        const wrapper = shallow(  < CounterApp value={ 105 } /> );

        wrapper.find('button').at(0).simulate('click');

        wrapper.find('button').at(1).simulate('click');
        
        const counter = wrapper.find('h2').text().trim();

        expect( counter ).toBe( '105' );
    });

});