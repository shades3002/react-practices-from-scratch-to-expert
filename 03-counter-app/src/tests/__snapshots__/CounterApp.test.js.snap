// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <CounterApp /> should show <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
